File: demo/examples/python/vector_examples/vector_create.py
import skypydb

# Create a client
client = skypydb.httpClient(
    api_url="http://localhost:8000",
    api_key="local-dev-key",
    embedding_provider="ollama",
    embedding_model_config={
        "model": "mxbai-embed-large",
        "base_url": "http://localhost:11434"
    }
)

# Create a vector database
vectordb = client.create_collection("my-videos")

File: demo/examples/js/vector_examples/vector_create.ts
import { vecClient } from "skypydb";

async function main(): Promise<void> {
  // Create a client
  const client = new vecClient({
    embedding_provider: "ollama",
    embedding_model_config: {
      model: "mxbai-embed-large",
      base_url: "http://localhost:11434",
    },
  });

  try {
    // Create a vector database
    const vectordb = await client.create_collection("my-videos");

    // Print the name of the created collection
    console.log(`Created collection: ${vectordb.name}`);
  } finally {
    // Close local resources
    await client.close();
  }
}
